test/benchmarks/utils.py

```diff
@@ -92,7 +92,7 @@ def index_to_doc_store(doc_store, docs, retriever, labels=None):
         doc_store.write_labels(labels, index=label_index)
     # these lines are not run if the docs.embedding field is already populated with precomputed embeddings
     # See the prepare_data() fn in the retriever benchmark script
-    elif callable(getattr(retriever, "embed_passages", None)) and docs[0].embedding is None:
+    if callable(getattr(retriever, "embed_passages", None)) and docs[0].embedding is None:
```

Comment:
@brandenchan could you have a look at this change, this might have caused a bug on your side. Im unsure about precomputed embeddings and how this factors in

Reply:
Did this change make any impact on your runs? I have had no reason to suspect that there is any issue with how embeddings are populated in the benchmarks so far. As far as I can see, I think the benchmarks will still run the same whether this is an `if` or an `elif`. But of course, we  should take on this change if you see the need for it.

My reasoning is as follows: during index benchmarking, there are no labels and update embeddings is called by this fn. During query benchmarking there are labels and embeddings are created in `prepare_data()`. These two cases are essentially covered by the conditional `if` `else` blocks that this change is altering.

Reply:
After discussing with Brranden using if seems correct for all cases. In the special case of dpr the questions were saved with pre computed embeddings. We of course want to be able to compute embeddings for queries when we attach other embedding models.